reflect.call error for include closure type in outputs

```
package main

import "reflect"

func demo1(fn func(n int) int) func(n int) int {
	return func(n int) int {
		return fn(n + 100)
	}
}

func main() {
	var base = 100
	fn := func(n int) int {
		return n + base
	}
	// demo1
	f1 := reflect.ValueOf(demo1)
	r1 := f1.Call([]reflect.Value{reflect.ValueOf(fn)})
	if r1[0].Call([]reflect.Value{reflect.ValueOf(100)})[0].Int() != 300 {
		panic("call demo1 error")
	}
}
```

llgo run .
```
panic: runtime error: invalid memory address or nil pointer dereference
```

--- a/.github/workflows/test_demo.sh
+++ b/.github/workflows/test_demo.sh
@@ -102,6 +102,7 @@ ignore_esp32=(
   "./_demo/go/randcrypt" # panic: internal/bytealg selected .s files require plan9asm translation
   "./_demo/go/randdemo" # panic: internal/bytealg selected .s files require plan9asm translation
   "./_demo/go/readdir" # panic: internal/bytealg selected .s files require plan9asm translation
+  "./_demo/go/reflectcallfn" # panic: internal/bytealg selected .s files require plan9asm translation
   "./_demo/go/reflectchanof" # panic: internal/bytealg selected .s files require plan9asm translation
   "./_demo/go/reflectconv" # panic: internal/bytealg selected .s files require plan9asm translation
   "./_demo/go/reflectcopy" # panic: internal/bytealg selected .s files require plan9asm translation
@@ -185,6 +186,7 @@ ignore_esp32c3_basic=(
   "./_demo/go/randcrypt" # panic: internal/bytealg selected .s files require plan9asm translation
   "./_demo/go/randdemo" # panic: internal/bytealg selected .s files require plan9asm translation
   "./_demo/go/readdir" # panic: internal/bytealg selected .s files require plan9asm translation
+  "./_demo/go/reflectcallfn" # panic: internal/bytealg selected .s files require plan9asm translation
   "./_demo/go/reflectchanof" # panic: internal/bytealg selected .s files require plan9asm translation
   "./_demo/go/reflectconv" # panic: internal/bytealg selected .s files require plan9asm translation
   "./_demo/go/reflectfunc" # panic: internal/bytealg selected .s files require plan9asm translation

--- a/_demo/go/reflectcallfn/main.go
+++ b/_demo/go/reflectcallfn/main.go
@@ -1,0 +1,81 @@
+package main
+
+import (
+	"reflect"
+)
+
+type M struct {
+	M1 func(int) int
+	M2 func(int) int
+}
+
+type N struct {
+	N1 func(int) int
+	N2 func(int) int
+	M  M
+}
+
+func demo2(fn func(n int) int) (func(int) int, func(int) int) {
+	return func(n int) int {
+			return fn(n + 100)
+		}, func(n int) int {
+			return fn(n + 200)
+		}
+}
+
+func demo1(fn func(n int) int) func(n int) int {
+	return func(n int) int {
+		return fn(n + 100)
+	}
+}
+
+func demo3(fn func(n int) int) N {
+	return N{
+		N1: func(n int) int { return fn(n) + 100 },
+		N2: func(n int) int { return fn(n) + 200 },
+		M: M{
+			M1: func(n int) int { return fn(n) + 300 },
+			M2: func(n int) int { return fn(n) + 400 },
+		},
+	}
+}
+
+func main() {
+	var base = 100
+	fn := func(n int) int {
+		return n + base
+	}
+	// demo1
+	f1 := reflect.ValueOf(demo1)
+	r1 := f1.Call([]reflect.Value{reflect.ValueOf(fn)})
+	if r1[0].Call([]reflect.Value{reflect.ValueOf(100)})[0].Int() != 300 {
+		panic("call demo1 error")
+	}
+	// demo2
+	f2 := reflect.ValueOf(demo2)
+	r2 := f2.Call([]reflect.Value{reflect.ValueOf(fn)})
+	if r2[0].Call([]reflect.Value{reflect.ValueOf(100)})[0].Int() != 300 {
+		panic("call demo2 error")
+	}
+	if r2[1].Call([]reflect.Value{reflect.ValueOf(100)})[0].Int() != 400 {
+		panic("call demo2 error")
+	}
+	// demo3
+	f3 := reflect.ValueOf(demo3)
+	r3 := f3.Call([]reflect.Value{reflect.ValueOf(fn)})
+	if r3[0].Field(0).Call([]reflect.Value{reflect.ValueOf(100)})[0].Int() != 300 {
+		panic("call N.N1 error")
+	}
+	if r3[0].Field(1).Call([]reflect.Value{reflect.ValueOf(100)})[0].Int() != 400 {
+		panic("call N.N2 error")
+	}
+	if r3[0].Field(2).Field(0).Call([]reflect.Value{reflect.ValueOf(100)})[0].Int() != 500 {
+		panic("call N.M.M1 error")
+	}
+	if r3[0].Field(2).Field(1).Call([]reflect.Value{reflect.ValueOf(100)})[0].Int() != 600 {
+		panic("call N.M.M2 error")
+	}
+	if r3[0].Field(2).Field(0).Call([]reflect.Value{reflect.ValueOf(100)})[0].Int() != 500 {
+		panic("call N.M.M1 error")
+	}
+}

--- a/runtime/internal/lib/reflect/value.go
+++ b/runtime/internal/lib/reflect/value.go
@@ -616,7 +616,9 @@ func (v Value) Field(i int) Value {
 
 	// Check closure to func
 	kind := typ.Kind()
-	if typ.IsClosure() {
+	if kind == abi.Func {
+		typ = closureOf(typ.FuncType())
+	} else if typ.IsClosure() {
 		kind = abi.Func
 	}
 	// Inherit permission bits from v, but clear flagEmbedRO.
@@ -2370,6 +2372,9 @@ func toFFIType(typ *abi.Type) *ffi.Type {
 	case abi.String:
 		return ffi.TypeString
 	case abi.Struct:
+		if typ.IsClosure() {
+			return ffiTypeClosure
+		}
 		return toFFIStructType(typ)
 	case abi.UnsafePointer:
 		return ffi.TypePointer
@@ -2565,6 +2570,7 @@ func (v Value) call(op string, in []Value) (out []Value) {
 		args = append(args, toFFIArg(arg, typ))
 	}
 
+	tout = toRuntimeTypes(tout)
 	sig, err := ffi.NewSignature(toFFIRetType(tout), ffiArgs...)
 	if err != nil {
 		panic(err)
@@ -2592,6 +2598,29 @@ func (v Value) call(op string, in []Value) (out []Value) {
 		}
 	}
 	return
+}
+
+func toRuntimeTypes(typs []*abi.Type) []*abi.Type {
+	switch n := len(typs); n {
+	case 0:
+		return nil
+	case 1:
+		return []*abi.Type{toRuntimeType(typs[0])}
+	default:
+		rets := make([]*abi.Type, n)
+		for i, typ := range typs {
+			rets[i] = toRuntimeType(typ)
+		}
+		return rets
+	}
+}
+
+func toRuntimeType(typ *abi.Type) *abi.Type {
+	switch typ.Kind() {
+	case abi.Func:
+		return closureOf((*abi.FuncType)(unsafe.Pointer(typ)))
+	}
+	return typ
 }
 
 // resolveIndirectValue converts Value's indirect reference to direct based on abi.Type, clears flagIndir

